### DISCUSSION TITLE
x-model not working when using x-spread

### DISCUSSION THREAD

**Discussion by romsar** [alpinejs/alpine#1069] [1. Help] [answered]
Hey guys!
I'm trying to learn that amazing framework.
Can you please explain why [this code](https://jsfiddle.net/rom4i/b10ukdfx/55/) is not working?

```
<div x-data="contactForm()">
  <form x-spread="submitForm">
	  Имя: <input x-spread="name" type="text" />
	  <input type="submit" value="Submit" />
	</form>
</div>
```

```
function contactForm() {
  return {
	  formData: {
		  name: '',
		},
		name: {
		  ['x-model']() {
			  return this.formData.name;
			},
		},
		submitForm: {
		  ['@submit.prevent']() {
				alert('Form submitted!');
			},
		},
	};
};
```

**Comment by ryangjchandler** [comment, marked as answer]
@RomanSarvarov You need to provide `x-model` with a string because of how Alpine generates the expression to re-assign the value after an update, please see here: https://jsfiddle.net/a3kbp4fn/

```js
function contactForm() {
    return {
        formData: {
            name: 'Testing',
        },
        name: {
            'x-model': 'formData.name',
        },
        submitForm: {
            ['@submit.prevent']() {
                console.log(this.formData.name);
                alert('Form submitted!');
            },
        },
    };
}
```

**Reply by romsar** [reply]
Thanks! I think this should be in documentation.

**Reply by ryangjchandler** [reply]
> Thanks! I think this should be in documentation.

Feel free to make a pull request! Also please mark this discussion as answered so that others can find it easier.

**Reply by romsar** [reply]
I'll be happy to make a PR, but my english is not good enough. I don't even know how to describe it :)
